### PR TITLE
Lab2: Expose isProjectTemplateLevel and show icon in AI Chat

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
## Warning!!

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2024. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.

## Description

Expose the `isProjectTemplateLevel` property in Lab2, and use that to show the ProjectTemplateWorkspaceIcon in the AI Chat header. The "projectTemplateLevelName" property comes automatically from the level's properties, so we just need to check if that property exists.

Related PR to enable project template backed levels for AI Chat: https://github.com/code-dot-org/code-dot-org/pull/57885

There's probably some future discussion here on if Lab2 should somehow control showing that icon automatically, but currently since all Labs own their own UI entirely, it felt like it made sense to have AI Chat just show the icon.

AI Chat edit page with icon:
Screenshot 2024-04-08 at 5 58 24 PM (2)

AI Chat presentation page with icon:
Screenshot 2024-04-08 at 5 58 57 PM (2)

Non-Lab2 lab with icon (default tooltip placement above icon):
Screenshot 2024-04-08 at 5 59 39 PM (2)

## Links

https://codedotorg.atlassian.net/browse/LABS-671

## Testing story

Tested with project template backed levels in Lab2 and outside of Lab2.

### CODE REVIEW

File: apps/src/templates/ProjectTemplateWorkspaceIcon.jsx [hunk 1]
```diff
@@ -28,6 +33,7 @@ export default class ProjectTemplateWorkspaceIcon extends React.Component {
           role="tooltip"
```
Comment (fisher-alice) [inline apps/src/templates/ProjectTemplateWorkspaceIcon.jsx:33]: Currently, the icon/tooltip is not tab-navigable. While you're here, what do you think of wrapping image in a button? With the version of `ReactTooltip` we're using `3.2.7` (kinda old), you could add the prop `data-event`. Here's an example: https://github.com/code-dot-org/code-dot-org/commit/fa62bd24c24efc9b80ace5b25118986c4b90f35d

Comment (fisher-alice) [reply apps/src/templates/ProjectTemplateWorkspaceIcon.jsx:33]: Nit: could migrate style to module if you make suggested update above.

File: apps/src/aichat/views/AichatView.tsx [hunk 2]
```diff
@@ -73,8 +78,14 @@ const AichatView: React.FunctionComponent = () => {
     onChange: setViewMode,
   };
 
-  const chatWorkspaceHeader =
-    viewMode === ViewMode.EDIT ? aichatI18n.aichatWorkspaceHeader() : botName;
+  const chatWorkspaceHeader = (
```
Comment (molly-moen) [inline apps/src/aichat/views/AichatView.tsx:80]: long term we could consider a common header component that checks the `projectTemplateLevel` value and adds the icon if necessary, but this seems reasonable for now!

**Review by molly-moen — APPROVED** [review]

File: apps/src/templates/ProjectTemplateWorkspaceIcon.jsx (hunk 1 above)
Comment (sanchitmalhotra126) [reply apps/src/templates/ProjectTemplateWorkspaceIcon.jsx:33]: Will do, thanks for the code snippet!

Edit: actually, I'll handle this in a follow-up to keep the changes separate.

**Review by fisher-alice — APPROVED** [review]
Nice work! Thanks for taking care of this so quickly.

**Pull request merged**